kdb/kdb.go

```diff
@@ -0,0 +1,151 @@
+package kdb
+
+// #cgo LDFLAGS: -lelektra
+// #include <elektra/kdb.h>
+import "C"
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// KDB is an interface to the elektra library.
+type KDB interface {
+	Open(key Key) error
+	Close(key Key) error
+
+	Get(keySet KeySet, parentKey Key) error
+	Set(keySet KeySet, parentKey Key) error
+
+	// Ensure(contract KeySet, parentKey Key)
+
+	Version() (string, error)
+}
+
+type kdbC struct {
+	handle *C.struct__KDB
+}
+
+// New returns a new KDB instance.
+func New() KDB {
+	return &kdbC{}
+}
+
+// Open creates a handle to the elektra library,
+// this is mandatory to Get / Set Keys.
+func (e *kdbC) Open(key Key) error {
+	k, err := toCKey(key)
+
+	if err != nil {
+		return err
+	}
+
+	handle, ret := C.kdbOpen(k.key)
+
+	if handle == nil {
+		return fmt.Errorf("unable to open kdb: %v", ret)
+	}
+
+	e.handle = handle
+
+	return nil
+}
+
+// Close closes the kdb handle.
+func (e *kdbC) Close(key Key) error {
+	ckey, err := toCKey(key)
+
+	if err != nil {
+		return err
+	}
+
+	ret := C.kdbClose(e.handle, ckey.key)
+
+	if ret < 0 {
+		return errors.New("could not close kdb handle")
+	}
+
+	return nil
+}
+
+// Get retrieves parentKey and all Keys beneath it.
+func (e *kdbC) Get(keySet KeySet, parentKey Key) error {
+	cKey, err := toCKey(parentKey)
+
+	if err != nil {
+		return err
+	}
+
+	cKeySet, err := toCKeySet(keySet)
+
+	if err != nil {
+		return err
+	}
+
+	C.kdbGet(e.handle, cKeySet.keySet, cKey.key)
+
+	return nil
```

Comment:
Please return the return value of kdbGet (0 and 1 says if something was done or if everything was already up-to-date).